dependencies.yaml

```diff
@@ -0,0 +1,124 @@
+# Dependency list for https://github.com/rapidsai/dependency-file-generator
+files:
+  all:
+    output: conda
+    matrix:
+      cuda: ["11.5"]
+      arch: [x86_64]
+    includes:
+      - build
+      - cudatoolkit
+      - develop
+      - doc
+      - notebook
+      - run
+      - test_python
+channels:
+  - rapidsai
+  - rapidsai-nightly
+  - dask/label/dev
+  - conda-forge
+  - nvidia
+dependencies:
+  build:
+    common:
+      - output_types: [conda, requirements]
+        packages:
+          - cmake>=3.23.1,!=3.25.0
+          - cuda-python >=11.7.1,<12.0
+          - cython>=0.29,<0.30
+          - ninja
+          - scikit-build>=0.13.1
+      - output_types: [conda]
+        packages:
+          - c-compiler
+          - cxx-compiler
+          - rapids-build-env=23.02.*
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              arch: x86_64
+            packages:
+              - gcc_linux-64=9.*
+              - sysroot_linux-64==2.17
+          - matrix:
+              arch: aarch64
+            packages:
+              - gcc_linux-aarch64=9.*
+              - sysroot_linux-aarch64==2.17
+  develop:
+    common:
+      - output_types: [conda, requirements]
+        packages:
+          - clang=11.1.0
```

Comment:
I think we can drop clang/clang-tools, which means the entire `develop` list could be deleted. I don't think these are used anywhere. clang-format (the original reason this was added, afaik) is now handled via pre-commit.

Reply:
I believe @cjnolet said that they do/plan to run `clang`.

I notice this script among others https://github.com/rapidsai/raft/blob/branch-23.02/cpp/scripts/run-clang-tidy.py

Reply:
I think only `clang-tools` is required for providing the `clang-tidy` binary. No scripts are calling `clang` as far as I can tell.

Reply:
We are in the process of updating our code so we can support clang. I'd prefer to leave in the dependency in the meantime (we do have an experimental clang compile script in there that we're working on making build successfully).

Reply:
Okay. We can resolve this.